modules/tinymce/src/plugins/media/test/ts/browser/SubmitTest.ts

```diff
@@ -1,82 +1,80 @@
-  Plugin();
-  Theme();
+describe('browser.tinymce.plugins.media.core.SubmitTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: [ 'media' ],
+    toolbar: 'media',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ]);
+
+  const mediaUrlResolver = (data: { url: string }, resolve: (data: { html: string }) => void) => {
+    Delay.setTimeout(() => {
+      resolve({
+        html: '<span id="fake">' + data.url + '</span>'
+      });
+    }, 500);
+  };
 
   const customEmbed =
   '<div style="left: 0px; width: 100%; height: 0px; position: relative; padding-bottom: 56.338%; max-width: 650px;">' +
   '<iframe src="https://www.youtube.com/embed/IcgmSRJHu_8"' +
   ' width="560" height="314" allowfullscreen="allowfullscreen" data-mce-fragment="1">' +
   '</iframe></div>';
 
-  const sTestResolvedEmbedContentSubmit = (ui: TinyUi, editor: Editor, apis: TinyApis, url: string, expected: string) =>
-    Logger.t(`Assert content ${expected}`, GeneralSteps.sequence([
-      Utils.sOpenDialog(ui),
-      Utils.sSetFormItemNoEvent(ui, url),
-      ui.sClickOnUi('click save button', Utils.selectors.saveButton),
-      Utils.sAssertEditorContent(apis, editor, expected)
-    ]));
+  const pTestResolvedEmbedContentSubmit = async (editor: Editor, url: string, expected: string) => {
+    await Utils.pOpenDialog(editor);
+    await Utils.pSetFormItemNoEvent(editor, url);
+    TinyUiActions.submitDialog(editor);
+    await Utils.pAssertEditorContent(editor, expected);
+  };
 
-  const sTestManualEmbedContentSubmit = (ui: TinyUi, editor: Editor, apis: TinyApis, embed: string, expected: string) =>
-    Logger.t(`Assert content ${expected}`, GeneralSteps.sequence([
-      Utils.sOpenDialog(ui),
-      Utils.sPasteTextareaValue(ui, embed),
-      ui.sClickOnUi('click save button', Utils.selectors.saveButton),
-      Utils.sAssertEditorContent(apis, editor, expected)
-    ]));
+  const pTestManualEmbedContentSubmit = async (editor: Editor, embed: string, expected: string) => {
+    await Utils.pOpenDialog(editor);
+    await Utils.pPasteTextareaValue(editor, embed);
+    TinyUiActions.submitDialog(editor);
+    await Utils.pAssertEditorContent(editor, expected);
+  };
 
-  const sTestEmbedUnchangedAfterOpenCloseDialog = (ui: TinyUi, editor: Editor, apis: TinyApis, expected: string) =>
-    GeneralSteps.sequence([
-      Utils.sOpenDialog(ui),
-      ui.sClickOnUi('click save button', Utils.selectors.saveButton),
-      Utils.sAssertEditorContent(apis, editor, expected)
-    ]);
+  const pTestEmbedUnchangedAfterOpenCloseDialog = async (editor: Editor, expected: string) => {
+    await Utils.pOpenDialog(editor);
+    TinyUiActions.submitDialog(editor);
+    await Utils.pAssertEditorContent(editor, expected);
+  };
 
-  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-    const ui = TinyUi(editor);
-    const apis = TinyApis(editor);
+  beforeEach(() => {
```

Comment:
Would that translate better as an `afterEach`?

Reply:
While it would translate better, we definitely want a `beforeEach` here as the code was trying to make sure the editor was back to the initial state, while always assuming the editor started enpty.